src/dsp_tools/analyse_xml_data/xml_to_graph_models.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+import uuid
+from dataclasses import dataclass
+from typing import Optional
+from uuid import UUID
+
+
+@dataclass
+class TripleGraph:
+    """
+    Contains all classes that express links between resources.
+    """
+
+    resptr_links: Optional[list[ResptrLink]]
+    xml_links: Optional[list[XMLLink]]
```

Comment:
`Optional[A]` can be expressed as `A | None` since some versions of Python. But in many cases, lists don't need to be optional anyways, as an empty list can have the same meaning as None.

Reply:
I chose optional because I would not assign a value at all if there are no xml links...

Reply:
In that case, the parameter can have the value `[]`, which works perfect, because `bool([]) == False`. It would make type hints easier

Reply:
No, that will work fine: `my_list: list[str] = []` is valid.

Reply:
Great thanks!